### ISSUE TITLE
[CLI] RuntimeError: OrderedDict mutated during iteration

### ISSUE THREAD

**Issue by konstantinjdobler** [wandb/client#2251]
I am getting a `RuntimeError: OrderedDict mutated during iteration` error whenever I try to do _any forward_ calls with a model that I have previously called `wandb.watch` on. **This does not happen when I do not call `wandb.watch` on the model**. I have manually tested all wandb versions from `0.10.27` and upwards and can confirm that this error only starts to appear in `0.10.30` and is not fixed in `0.10.31`.

In this specific traceback I am also using PytorchLightning, however I was getting the same error in an other project without Lightning. I have been forced to downgrade `wandb` to `0.10.29` due to this.

Traceback:
```sh
Traceback (most recent call last):
  File "src/gan/train.py", line 83, in <module>
    trainer.fit(model, dm)
  File "/Users/user/prog/mp/genart/venv/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 458, in fit
    self._run(model)
 <... a bunch of irrelevant PytorchLightning stuff ...>
  File "/Users/user/prog/mp/genart/venv/lib/python3.7/site-packages/pytorch_lightning/core/memory.py", line 242, in summarize
    self._forward_example_input()
  File "/Users/user/prog/mp/genart/venv/lib/python3.7/site-packages/pytorch_lightning/core/memory.py", line 267, in _forward_example_input
    model(input_)
  File "/Users/user/prog/mp/genart/venv/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/Users/user/prog/mp/genart/src/gan/outer_gan.py", line 127, in forward
    generated_images = self.generator(z, labels)
  File "/Users/user/prog/mp/genart/venv/lib/python3.7/site-packages/torch/nn/modules/module.py", line 892, in _call_impl
    self._forward_hooks.values()):
RuntimeError: OrderedDict mutated during iteration
```

**Comment by cvphelps** [comment]
Hi @konstantinjdobler thanks for reporting this. @borisdayma could you please take a look.

**Comment by epwalsh** [comment]
I'm seeing this as well using our W&B training callback in AllenNLP with `watch_model=True`.

![image](https://user-images.githubusercontent.com/8812459/121072347-f3375680-c785-11eb-8449-ff4bfdd72fad.png)

**Comment by yqsony** [comment]
We had a same issue when running in 0.10.30. We had to revert back to an earlier version.

**Comment by borisdayma** [comment]
Just letting you know it's being worked on. Hopefully we'll have a fix for next week.

**Comment by borisdayma** [comment]
To all, I didn't have your specific example so could not test against it.

Feel free to check against PR #2271 to see if this fixes your issue :crossed_fingers:

**Comment by vanpelt** [comment]
Hey everyone,

We just released 0.10.32 of our library that should address this issue.  I'm going to close this issue for now but please feel free to add comments with any errors you continue to see with the most recent release of the library.
